login.go

```diff
@@ -39,7 +40,7 @@ func (c *LoginController) Authorize(ctx *app.AuthorizeLoginContext) error {
 
 // Generate runs the authorize action.
 func (c *LoginController) Generate(ctx *app.GenerateLoginContext) error {
-	if !Development {
+	if !viper.GetBool("developer.mode.enabled") {
```

Comment:
I would prefer to keep the knowledge about which configuration options we have (and how we implement configuration) inside the configuration package. Explicitly, I would offer an API like so:

configuration.isDeveloperModeEnabled()

The names of the config variables and the viper references seems like implementation details. What's the story for programmatically changing the setup vars, for example in tests?

This comment applies to all places we read setup info further down, as well.